filter_xml/data_handler.py

```diff
@@ -13,14 +17,22 @@ def __init__(self, *args, **kwargs) -> None:
         smiley_file = kwargs.pop('file', None)
 
         self.smiley_file = smiley_file if smiley_file else self.SMILEY_XML
-        self.should_get_xml = False if smiley_file else True
+        self.should_get_xml = False if smiley_file else is_file_from_today(self.SMILEY_XML)
```

Comment:
Wouldn't this result in downloading a new XML file if it's from today?
Shouldn't it be `False if smiley_file or is_file_from_today(self.SMILEY_XML)`

Reply:
Actually `should_get_xml` is only used if the file is not from today so shouldn't this variable be moved to above line `32`?

Reply:
You're right - I flipped it.

Fixed in 7709250 and 7a0d69c

`should_get_xml` is used if the file is from today and if we pass `--file, -f`